pkg/registry/pod/rest.go

```diff
@@ -59,6 +59,14 @@ func (podStrategy) ResetBeforeCreate(obj runtime.Object) {
 	}
 }
 
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+func (podStrategy) PrepareForUpdate(obj, old runtime.Object) {
```

Comment:
Another great observation.  This REST code has changed a lot since I touched it last :)

What's the clean fix here?  Just override PrepareForUpdate on StatusStrategy?